### DISCUSSION TITLE
How to make relation object required !

### DISCUSSION THREAD

**Discussion by afcode123** [prisma/prisma#9706] [Q&A]
```prisma
model User {
  id   Int    @id @default(autoincrement())
  post Post[]
}

model Post {
  id     Int  @id @default(autoincrement())
  user   User @relation("PostToUser", fields: [userId], references: [id])
  userId Int
}
```
Question: With such example relation model userId in Post model becomes required when running prisma generate. However it does not seem to make "user" field required in Post. Is it possible to define the model so that field "user" can not be undefined for post?

**Comment by ryands17** [comment]
@afcode123 👋 
This required relation will make sure that `User` is not undefined for `Post`. In what case are you getting an undefined?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
